biooutput.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+import numpy as np
+
+class Biooutput():
+    """Output Class
+
+    """
+
+    def __init__(self, hr_data3):
+        self.data = hr_data3
+
+    def summarize_ecg(self):
+        """Create txt file summarizing ECG analysis
+
+        :param data: pandas dataframe {'Heartrate', 'B\T', 'time'}
+        :return ecgResults.txt
+        """
+
+        # Finds the last time point in the data file
+        maxTime = self.data['time'].iat[-1]
+
+        # Tells user the purpose of this program
+        printStr = ("This program will summarize the ECG data.\n\
+        Data recorded from 0s to {}s.\n\
+        Enter a start and stop time in that range to get an instantaneous \n\
+        heart rate at those times, an average rate over that period, \n\
+        and any bradycardia or tachycardia events between those times.\n\n".format(maxTime))
+        print(printStr)
+
+        # Error messages for possible errors
+        errmsg_max = ('\nThe time you selected is longer than the recorded data!\n\
+        Please choose between 0s and {}s\n\n'.format(maxTime))
+        errmsg_notint = ('\nNot an integer! Please choose a time between 0s and {}s\n\n'.format(maxTime))
+
+        # Get user specified period for averaging heartrate
+        startTimestr = 'Start time (in seconds): '
+
+        # Check for valid input
+        while True:
+            try:
+                startTime = int(input(startTimestr))
+            except ValueError:
+                print(errmsg_notint)
+            else:
```

Comment:
this logic is a bit unusual (personally I'm not a huge fan of infinite loops in a program, usually it's imperative to make the logic airtight and clear in such cases). I think what you want to do here is skip the current loop iteration if a `ValueError` is thrown right? In that case you can use `continue` to skip the current loop iteration.

```python
except ValueError:
    print(...)
    continue
```

Reply:
Fixed, added the continue after the print statement.